src/librustc_mir/borrow_check/conflict_errors.rs

```diff
@@ -595,12 +595,11 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     ) -> (String, String, String, String) {
         // Define a small closure that we can use to check if the type of a place
         // is a union.
-        let is_union = |place: &Place<'tcx>| -> bool {
-            place.ty(self.mir, self.infcx.tcx).ty
-                .ty_adt_def()
-                .map(|adt| adt.is_union())
-                .unwrap_or(false)
+        let is_union = |place: &Place<'tcx>| -> Option<Ty<'tcx>> {
```

Comment:
The name of this lambda is suboptimal now that it doesn't return `bool`.

Reply:
Done!